iTwin/itwinjs-core
# Add EXT_mesh_features support

This PR introduces support for reading and interpreting Cesium tiles with [`EXT_mesh_features` extension](https://github.com/CesiumGS/glTF/blob/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features/README.md). Support only extends to tilesets that hold [Feature ID information by Vertex](https://github.com/CesiumGS/glTF/blob/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features/README.md#feature-id-by-vertex) and will ignore Feature ID by Texture Coordinates.

This feature was requested here: https://github.com/iTwin/itwin-graphics-backlog/issues/425

The new behavior can be tested by:
1. Serve Cesium tileset that has `EXT_mesh_features` on localhost
2. Run display-test-app
3. Open a blank connection
4. In the integrated console enter `fdt attach reality model <path to served tileset.json>`
5. If the tileset is not visible, try pressing `Fit to View` button
6. Hover over some visible object - a tooltip with object metadata should appear.

![ext_mesh_features](https://github.com/user-attachments/assets/257d1ba5-6ce1-4979-b9be-4fd73c3fc251)
Model used in this image was also used in GltfReader tests.
